scripts/globals/survival_guide.lua

```diff
@@ -0,0 +1,136 @@
+
+        local updatedValue = mask + guide.groupMask
+
+        if (variableSuffix == 'a') then
+            updatedValue = guide.groupMask
+            masks[group + 4] = updatedValue
+        else
+            masks[group] = updatedValue
+        end
+
+        player:setVar("SgTeleportMask" .. group .. variableSuffix, updatedValue)
+    end
+
+    return masks
+end
+
+-------------------------------------------------
+-- public functions
+-------------------------------------------------
+
+dsp.survival_guide.onTrigger = function(player, csid, index)
+    local guide = survivalGuides[index]
+
+    if (not (guide == nil) or not (ENABLE_ROV == 1)) then
+
+        -- Contains the zone to include, this is a bit-wise value.
+        local expansions = getEnabledExpansions()
+
+        local masks = {
+            [1] = player:getVar("SgTeleportMask1"),
+            [2] = player:getVar("SgTeleportMask2"),
+            [3] = player:getVar("SgTeleportMask3"),
+            [4] = player:getVar("SgTeleportMask4"),
+            [5] = player:getVar("SgTeleportMask1a"),
+            [6] = player:getVar("SgTeleportMask2a"),
+            [7] = player:getVar("SgTeleportMask3a")
+        }
+        -- If this survival guide hasn't been registered yet (saved to database) do that now.
+        masks = checkForRegisteredSurvivalGuide(player, guide, masks)
+
+        -- param 1 = Favorites, this is broken yet, not sure how to fix this, don't really care right now either, the rest of it works (except for tabs).
+        -- param 2 = current area - this is the zone index passed in, this means that the area won't show in the list of areas to teleport too.
+        -- param 3 = gil, NOTE: The startEvent method probably needs to be updated to allow for a table so gil and valor_point (tabs) can be passed in too.
+        -- param 4 = zones unlocked (group 1), set to -1 to enable all zones in the group.
+        -- param 5 = Zones unlocked (group 2), set to -1 to enable all zones in the group.
+        -- param 6 = Zones unlocked (group 3), set to -1 to enable all zones in the group.
+        -- param 7 = zones unlocked (Zehrun mines and Eastern Adoulin), set to -1 to enable all zones in the group.
+        -- param 8 = expansions available.
+        -- NOTE: To allow teleportation to all zones, set all of the mask values to -1.
+        player:startEvent(csid, -1, index, player:getGil(), masks[1] + masks[5],
+                          masks[2] + masks[6], masks[3] + masks[7], masks[4],
+                          expansions)
+    else
+        player:PrintToPlayer('Survival guides are not enabled!')
+    end
+
+    return 1
+end
+
+dsp.survival_guide.onEventFinish = function(player, csid, eventId, option)
+    if (option >= 1 and not (option == 7) and not (option == 65539) and csid ==
+        eventId) then
+        local selectedMenuId = bit.rshift(option, 16)
+
+        if (selectedMenuId <= 97) then
+            local guide = survivalGuides[selectedMenuId]
```

Comment:
If using an optionToID translation table (which I greatly advocate, this becomes `local guide = survivalGuides[optionToGude[option]]

Reply:
Updated how this is keyed